### ISSUE TITLE
Issues after upgrade from 2.9 to 2.10 and 42.3 to 15.1 in one go

### ISSUE THREAD

**Issue by frispete** [openSUSE/open-build-service#7912]
After $subject and following https://github.com/openSUSE/open-build-service/blob/2.10/dist/README.UPDATERS, I'm facing some issues here:

Some OBS services (publisher, warden and dodup) are shown as dead in /monitor after reset, although they appear as running for systemd:
```
$ systemctl status obspublisher.service obswarden.service obsdodup.service
* obspublisher.service - OBS repository publisher
   Loaded: loaded (/usr/lib/systemd/system/obspublisher.service; enabled; vendor preset: disabled)
   Active: active (running) since Mon 2019-07-15 16:23:13 UTC; 58s ago
 Main PID: 1813 (bs_publish)
    Tasks: 1 (limit: 4915)
   CGroup: /system.slice/obspublisher.service
           `-1813 /usr/bin/perl -w /usr/lib/obs/server/bs_publish --logfile publisher.log

Jul 15 16:23:13 obsserver systemd[1]: Started OBS repository publisher.

* obswarden.service - OBS warden, monitors the workers
   Loaded: loaded (/usr/lib/systemd/system/obswarden.service; enabled; vendor preset: disabled)
   Active: active (running) since Mon 2019-07-15 16:23:14 UTC; 57s ago
 Main PID: 1817 (bs_warden)
    Tasks: 1 (limit: 4915)
   CGroup: /system.slice/obswarden.service
           `-1817 /usr/bin/perl -w /usr/lib/obs/server/bs_warden --logfile warden.log

Jul 15 16:23:14 obsserver systemd[1]: Started OBS warden, monitors the workers.

* obsdodup.service - OBS dodup, updates download on demand metadata
   Loaded: loaded (/usr/lib/systemd/system/obsdodup.service; enabled; vendor preset: disabled)
   Active: active (running) since Mon 2019-07-15 16:23:13 UTC; 58s ago
 Main PID: 1812 (bs_dodup)
    Tasks: 1 (limit: 4915)
   CGroup: /system.slice/obsdodup.service
           `-1812 /usr/bin/perl -w /usr/lib/obs/server/bs_dodup --logfile dodup.log

Jul 15 16:23:13 obsserver systemd[1]: Started OBS dodup, updates download on demand metadata.
```

If I restart those 3 services manually, they're shown as running again (until next reboot).

Also, some running (during reboot) builders on the same host are not accessible after reboot. They're shown in /monitor{,/old}, but accessing the build log results in:
```
No live log available: connect to 127.0.0.1:37754: Connection refused
```
This doesn't happen for all jobs, just a few. Work around: toggle build flag of the affected package.

**Comment by adrianschroeter** [comment]
On Montag, 15. Juli 2019, 18:58:41 CEST Hans-Peter Jansen wrote:
> After $subject and following
> https://github.com/openSUSE/open-build-service/blob/2.10/dist/README.UPDATE
> RS, I'm facing some issues here:

Can you check if they appear as online in the backend via calling


 osc api /build/_workerstatus

?


> Some OBS services (publisher, warden and dodup) are shown as dead in
> /monitor after reset, although they appear as running for systemd: ```
> $ systemctl status obspublisher.service obswarden.service obsdodup.service
> * obspublisher.service - OBS repository publisher
>    Loaded: loaded (/usr/lib/systemd/system/obspublisher.service; enabled;
> vendor preset: disabled) Active: active (running) since Mon 2019-07-15
> 16:23:13 UTC; 58s ago Main PID: 1813 (bs_publish)
>     Tasks: 1 (limit: 4915)
>    CGroup: /system.slice/obspublisher.service
>            `-1813 /usr/bin/perl -w /usr/lib/obs/server/bs_publish --logfile
> publisher.log
> 
> Jul 15 16:23:13 obsserver systemd[1]: Started OBS repository publisher.
> 
> * obswarden.service - OBS warden, monitors the workers
>    Loaded: loaded (/usr/lib/systemd/system/obswarden.service; enabled;
> vendor preset: disabled) Active: active (running) since Mon 2019-07-15
> 16:23:14 UTC; 57s ago Main PID: 1817 (bs_warden)
>     Tasks: 1 (limit: 4915)
>    CGroup: /system.slice/obswarden.service
>            `-1817 /usr/bin/perl -w /usr/lib/obs/server/bs_warden --logfile
> warden.log
> 
> Jul 15 16:23:14 obsserver systemd[1]: Started OBS warden, monitors the
> workers.
> 
> * obsdodup.service - OBS dodup, updates download on demand metadata
>    Loaded: loaded (/usr/lib/systemd/system/obsdodup.service; enabled; vendor
> preset: disabled) Active: active (running) since Mon 2019-07-15 16:23:13
> UTC; 58s ago Main PID: 1812 (bs_dodup)
>     Tasks: 1 (limit: 4915)
>    CGroup: /system.slice/obsdodup.service
>            `-1812 /usr/bin/perl -w /usr/lib/obs/server/bs_dodup --logfile
> dodup.log
> 
> Jul 15 16:23:13 obsserver systemd[1]: Started OBS dodup, updates download on
> demand metadata. ```
> 
> If I restart those 3 services manually, they're shown as running again
> (until next reboot).
> 
> Also, some running (during reboot) builders on the same host are not
> accessible after reboot. They're shown in /monitor{,/old}, but accessing
> the build log results in: ```
> No live log available: connect to 127.0.0.1:37754: Connection refused
> ```
> This doesn't happen for all jobs, just a few. Work around: toggle build flag
> of the affected package.


-- 

Adrian Schroeter
SUSE Linux Products GmbH,  Maxfeldstr. 5, 90409 Nuernberg, Germany
email: adrian@suse.de

**Comment by seife** [comment]
> Some OBS services (publisher, warden and dodup) are shown as dead in /monitor after reset, although they appear as running for systemd:
...
> If I restart those 3 services manually, they're shown as running again (until next reboot).

where is your /srv/obs stored? Local disk or maybe NFS?

**Comment by frispete** [comment]
> Can you check if they appear as online in the backend via calling `osc api /build/_workerstatus`? 

After reset:
```
<workerstatus clients="8">
  <idle workerid="obsserver:1" hostarch="x86_64"/>
  <idle workerid="obsserver:2" hostarch="x86_64"/>
  <idle workerid="obsserver:3" hostarch="x86_64"/>
  <idle workerid="obsserver:4" hostarch="x86_64"/>
  <idle workerid="obsserver:5" hostarch="x86_64"/>
  <idle workerid="obsserver:6" hostarch="x86_64"/>
  <idle workerid="obsserver:7" hostarch="x86_64"/>
  <idle workerid="obsserver:8" hostarch="x86_64"/>
  <waiting arch="armv7l" jobs="0"/>
  <waiting arch="i586" jobs="0"/>
  <waiting arch="x86_64" jobs="0"/>
  <blocked arch="armv7l" jobs="0"/>
  <blocked arch="i586" jobs="0"/>
  <blocked arch="x86_64" jobs="0"/>
  <buildavg arch="armv7l" buildavg="1200"/>
  <buildavg arch="i586" buildavg="1200"/>
  <buildavg arch="x86_64" buildavg="1200"/>
  <partition>
    <daemon type="srcserver" state="running" starttime="1563261418"/>
    <daemon type="deltastore" state="running" starttime="1563261417"/>
    <daemon type="service" state="running" starttime="1563261416"/>
    <daemon type="scheduler" arch="armv7l" state="running" starttime="1563261441">
      <queue high="0" med="0" low="0" next="0"/>
    </daemon>
    <daemon type="scheduler" arch="i586" state="running" starttime="1563261441">
      <queue high="0" med="0" low="0" next="0"/>
    </daemon>
    <daemon type="scheduler" arch="x86_64" state="running" starttime="1563261441">
      <queue high="0" med="0" low="0" next="0"/>
    </daemon>
    <daemon type="repserver" state="running" starttime="1563261418"/>
    <daemon type="dispatcher" state="running" starttime="1563261441"/>
    <daemon type="publisher" state="dead"/>
    <daemon type="signer" state="running" starttime="1563261419"/>
    <daemon type="warden" state="dead"/>
    <daemon type="dodup" state="dead"/>
  </partition>
</workerstatus>
```
After `systemctl restart obspublisher.service obswarden.service obsdodup.service`:
```
<workerstatus clients="8">
  <idle workerid="obsserver:1" hostarch="x86_64"/>
  <idle workerid="obsserver:2" hostarch="x86_64"/>
  <idle workerid="obsserver:3" hostarch="x86_64"/>
  <idle workerid="obsserver:4" hostarch="x86_64"/>
  <idle workerid="obsserver:5" hostarch="x86_64"/>
  <idle workerid="obsserver:6" hostarch="x86_64"/>
  <idle workerid="obsserver:7" hostarch="x86_64"/>
  <idle workerid="obsserver:8" hostarch="x86_64"/>
  <waiting arch="armv7l" jobs="0"/>
  <waiting arch="i586" jobs="0"/>
  <waiting arch="x86_64" jobs="0"/>
  <blocked arch="armv7l" jobs="0"/>
  <blocked arch="i586" jobs="0"/>
  <blocked arch="x86_64" jobs="93"/>
  <buildavg arch="armv7l" buildavg="1200"/>
  <buildavg arch="i586" buildavg="1200"/>
  <buildavg arch="x86_64" buildavg="1200"/>
  <partition>
    <daemon type="srcserver" state="running" starttime="1563261418"/>
    <daemon type="deltastore" state="running" starttime="1563261417"/>
    <daemon type="service" state="running" starttime="1563261416"/>
    <daemon type="scheduler" arch="armv7l" state="running" starttime="1563261441">
      <queue high="0" med="0" low="0" next="0"/>
    </daemon>
    <daemon type="scheduler" arch="i586" state="running" starttime="1563261441">
      <queue high="0" med="0" low="0" next="0"/>
    </daemon>
    <daemon type="scheduler" arch="x86_64" state="running" starttime="1563261441">
      <queue high="0" med="0" low="2" next="0"/>
    </daemon>
    <daemon type="repserver" state="running" starttime="1563261418"/>
    <daemon type="dispatcher" state="running" starttime="1563261441"/>
    <daemon type="publisher" state="running" starttime="1563261575"/>
    <daemon type="signer" state="running" starttime="1563261419"/>
    <daemon type="warden" state="running" starttime="1563261575"/>
    <daemon type="dodup" state="running" starttime="1563261575"/>
  </partition>
</workerstatus>
```

> where is your /srv/obs stored? Local disk or maybe NFS?

All local:
```
$ df -hT
Filesystem                    Type      Size  Used Avail Use% Mounted on
devtmpfs                      devtmpfs   63G     0   63G   0% /dev
tmpfs                         tmpfs      63G     0   63G   0% /dev/shm
tmpfs                         tmpfs      63G  9.2M   63G   1% /run
tmpfs                         tmpfs      63G     0   63G   0% /sys/fs/cgroup
/dev/sda1                     ext4       32G  4.2G   26G  14% /
/dev/mapper/OBS-server        ext4      504G   45G  458G   9% /srv/obs
/dev/mapper/OBS-cache         ext4       25G   45M   24G   1% /var/cache/obs/worker
/dev/mapper/OBS-worker_root_1 ext4      122G   61M  116G   1% /var/cache/obs/worker/root_1
/dev/mapper/OBS-worker_root_2 ext4      122G   61M  116G   1% /var/cache/obs/worker/root_2
/dev/mapper/OBS-worker_root_3 ext4      122G   61M  116G   1% /var/cache/obs/worker/root_3
/dev/mapper/OBS-worker_root_4 ext4      122G   61M  116G   1% /var/cache/obs/worker/root_4
/dev/mapper/OBS-worker_root_5 ext4      122G   61M  116G   1% /var/cache/obs/worker/root_5
/dev/mapper/OBS-worker_root_6 ext4      122G   61M  116G   1% /var/cache/obs/worker/root_6
/dev/mapper/OBS-worker_root_7 ext4      122G   61M  116G   1% /var/cache/obs/worker/root_7
/dev/mapper/OBS-worker_root_8 ext4      122G   61M  116G   1% /var/cache/obs/worker/root_8
tmpfs                         tmpfs      13G     0   13G   0% /run/user/0
```

**Comment by seife** [comment]
OK. I have the problem with /srv/obs on NFS that the services are started before it is already mounted ;-)

You can still try this:

```
obs:/etc/systemd/system # cat obsdeltastore.service.d/10-srv-obs.conf
[Unit]
RequiresMountsFor=/srv/obs
obs:/etc/systemd/system # find . -type l -name 10-srv-obs.conf |xargs ls -l
lrwxrwxrwx 1 root root 42 Jul 15 07:06 ./obs-api-support.target.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 15 07:02 ./obs-clockwork.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 15 07:06 ./obs-delayedjob-queue-consistency_check.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 15 07:06 ./obs-delayedjob-queue-default.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 15 07:06 ./obs-delayedjob-queue-issuetracking.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 15 07:06 ./obs-delayedjob-queue-mailers.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 15 07:06 ./obs-delayedjob-queue-project_log_rotate.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 15 07:06 ./obs-delayedjob-queue-quick@0.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 15 07:06 ./obs-delayedjob-queue-quick@1.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 15 07:06 ./obs-delayedjob-queue-quick@2.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 15 07:06 ./obs-delayedjob-queue-releasetracking.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 15 07:06 ./obs-delayedjob-queue-staging.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Aug 25  2017 ./obsdispatcher.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Aug 25  2017 ./obsdodup.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Aug 25  2017 ./obspublisher.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Aug 25  2017 ./obsrepserver.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Aug 25  2017 ./obsscheduler.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Aug 25  2017 ./obsservice.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Aug 25  2017 ./obssignd.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Aug 25  2017 ./obssigner.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 15 07:06 ./obs-sphinx.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Aug 25  2017 ./obssrcserver.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Aug 25  2017 ./obswarden.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
```

This ensures all obs services are only started affter /srv/obs is mounted.
Or just check in "journalctl -b" that srv-obs.mount unit is actually finished before any obs services are started.

**Comment by frispete** [comment]
Thanks, Stefan, for your support. 

Yes, that's the problem, but even after setting up, what you proposed:
```
obsserver:/etc/systemd/system# cat obsdeltastore.service.d/10-srv-obs.conf
[Unit]
RequiresMountsFor=/srv/obs
obsserver:/etc/systemd/system# find . -type l -name 10-srv-obs.conf |xargs ls -l
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obs-api-support.target.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obs-clockwork.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obs-delayedjob-queue-consistency_check.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obs-delayedjob-queue-default.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obs-delayedjob-queue-issuetracking.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obs-delayedjob-queue-mailers.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obs-delayedjob-queue-project_log_rotate.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obs-delayedjob-queue-quick@.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obs-delayedjob-queue-releasetracking.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obs-delayedjob-queue-staging.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obs-sphinx.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obsapisetup.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obsdispatcher.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obsdodup.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obsgetbinariesproxy.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obsnotifyforward.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obspublisher.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obsredis.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obsrepserver.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obsscheduler.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obsservice.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obsservicedispatch.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obssigner.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obssrcserver.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obswarden.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
lrwxrwxrwx 1 root root 42 Jul 16 08:16 ./obsworker.service.d/10-srv-obs.conf -> ../obsdeltastore.service.d/10-srv-obs.conf
```
(iow: everything obs* but obsstoragesetup should wait for /srv/obs), it still fails.

srv-obs.mount is done somehow:
```
obsserver:~# systemctl status srv-obs.mount
* srv-obs.mount - /srv/obs
   Loaded: loaded (/proc/self/mountinfo)
   Active: active (mounted) since Tue 2019-07-16 08:34:07 UTC; 7min ago
    Where: /srv/obs
     What: /dev/mapper/OBS-server
    Tasks: 0 (limit: 4915)
   CGroup: /system.slice/srv-obs.mount
```
As you expected, too late:
```
Jul 16 08:34:04 obsserver systemd[1]: Started OBS repository publisher.
Jul 16 08:34:04 obsserver systemd[1]: Started OBS dodup, updates download on demand metadata.
Jul 16 08:34:04 obsserver systemd[1]: Starting OBS storage setup...
[...]
Jul 16 08:34:07 obsserver kernel: sd 0:0:1:0: [sdb] Very big device. Trying to use READ CAPACITY(16).
Jul 16 08:34:07 obsserver obsstoragesetup[1818]: Looking for existing OBS Server LVM Volume
Jul 16 08:34:07 obsserver obsstoragesetup[1818]: Found BSConfig.pm
Jul 16 08:34:07 obsserver kernel: EXT4-fs (dm-0): mounted filesystem with ordered data mode. Opts: (null)
```
My humble guess, systemd cannot serialize by /srv/obs, because it doesn't know about it:
```
obsserver:~# cat /etc/fstab 
/dev/sda1 / ext4 defaults 1 1
```
Okay, easy enough:
```
obsserver:/etc/systemd/system# cat obsdeltastore.service.d/10-srv-obs.conf
[Unit]
After=obsstoragesetup.service
```
@adrianschroeter @seife is right, the all services must be delayed after storage is set up properly. It gets interesting, if non local storage has to be accounted for (as in @seife's case), though. 

One off, one rest.
```
<workerstatus clients="12">
  <idle workerid="obsserver:1" hostarch="x86_64"/>
  <idle workerid="obsserver:2" hostarch="x86_64"/>
  <idle workerid="obsserver:3" hostarch="x86_64"/>
  <idle workerid="obsserver:4" hostarch="x86_64"/>
  <idle workerid="obsserver:5" hostarch="x86_64"/>
  <idle workerid="obsserver:6" hostarch="x86_64"/>
  <idle workerid="obsserver:7" hostarch="x86_64"/>
  <idle workerid="obsserver:8" hostarch="x86_64"/>
  <building workerid="obsserver:1" hostarch="x86_64" project="LISA:Kernel" repository="openSUSE_13.2" package="icu" arch="x86_64" starttime="1563264253"/>
  <building workerid="obsserver:4" hostarch="x86_64" project="LISA:Essential" repository="openSUSE_13.2" package="libxml2" arch="x86_64" starttime="1563264935"/>
  <building workerid="obsserver:5" hostarch="x86_64" project="LISA:Essential" repository="openSUSE_13.2" package="gettext-runtime-mini" arch="x86_64" starttime="1563264935"/>
  <building workerid="obsserver:7" hostarch="x86_64" project="LISA:Essential" repository="openSUSE_13.2" package="gettext-runtime" arch="x86_64" starttime="1563264935"/>
  <waiting arch="armv7l" jobs="0"/>
  <waiting arch="i586" jobs="0"/>
  <waiting arch="x86_64" jobs="0"/>
  <blocked arch="armv7l" jobs="0"/>
  <blocked arch="i586" jobs="0"/>
  <blocked arch="x86_64" jobs="92"/>
  <buildavg arch="armv7l" buildavg="1200"/>
  <buildavg arch="i586" buildavg="1200"/>
  <buildavg arch="x86_64" buildavg="1200"/>
  <partition>
    <daemon type="srcserver" state="running" starttime="1563266975"/>
    <daemon type="deltastore" state="running" starttime="1563266971"/>
    <daemon type="service" state="running" starttime="1563266971"/>
    <daemon type="scheduler" arch="armv7l" state="running" starttime="1563267011">
      <queue high="0" med="0" low="0" next="0"/>
    </daemon>
    <daemon type="scheduler" arch="i586" state="running" starttime="1563267011">
      <queue high="0" med="0" low="0" next="0"/>
    </daemon>
    <daemon type="scheduler" arch="x86_64" state="running" starttime="1563267011">
      <queue high="0" med="0" low="2" next="0"/>
    </daemon>
    <daemon type="repserver" state="running" starttime="1563266975"/>
    <daemon type="dispatcher" state="running" starttime="1563267010"/>
    <daemon type="publisher" state="running" starttime="1563266974"/>
    <daemon type="signer" state="running" starttime="1563266976"/>
    <daemon type="warden" state="running" starttime="1563266971"/>
    <daemon type="dodup" state="running" starttime="1563266972"/>
  </partition>
</workerstatus>
```

According to the worker status, workers 1, 4, 5, and 7 are busy,  (I've rebooted the system a couple of times during fiddling with the above), but in reality:
```
l /var/cache/obs/worker/root_*/              
/var/cache/obs/worker/root_1/:
total 24
drwxr-xr-x  3 root root  4096 Jul 16 08:49 ./
drwxr-xr-x 12 root root  4096 Jul 16 08:49 ../
drwx------  2 root root 16384 Jul 16 08:49 lost+found/

/var/cache/obs/worker/root_2/:
total 24
drwxr-xr-x  3 root root  4096 Jul 16 08:49 ./
drwxr-xr-x 12 root root  4096 Jul 16 08:49 ../
drwx------  2 root root 16384 Jul 16 08:49 lost+found/

/var/cache/obs/worker/root_3/:
total 24
drwxr-xr-x  3 root root  4096 Jul 16 08:49 ./
drwxr-xr-x 12 root root  4096 Jul 16 08:49 ../
drwx------  2 root root 16384 Jul 16 08:49 lost+found/

/var/cache/obs/worker/root_4/:
total 24
drwxr-xr-x  3 root root  4096 Jul 16 08:49 ./
drwxr-xr-x 12 root root  4096 Jul 16 08:49 ../
drwx------  2 root root 16384 Jul 16 08:49 lost+found/

/var/cache/obs/worker/root_5/:
total 24
drwxr-xr-x  3 root root  4096 Jul 16 08:49 ./
drwxr-xr-x 12 root root  4096 Jul 16 08:49 ../
drwx------  2 root root 16384 Jul 16 08:49 lost+found/

/var/cache/obs/worker/root_6/:
total 24
drwxr-xr-x  3 root root  4096 Jul 16 08:49 ./
drwxr-xr-x 12 root root  4096 Jul 16 08:49 ../
drwx------  2 root root 16384 Jul 16 08:49 lost+found/

/var/cache/obs/worker/root_7/:
total 24
drwxr-xr-x  3 root root  4096 Jul 16 08:49 ./
drwxr-xr-x 12 root root  4096 Jul 16 08:49 ../
drwx------  2 root root 16384 Jul 16 08:49 lost+found/

/var/cache/obs/worker/root_8/:
total 24
drwxr-xr-x  3 root root  4096 Jul 16 08:49 ./
drwxr-xr-x 12 root root  4096 Jul 16 08:49 ../
drwx------  2 root root 16384 Jul 16 08:49 lost+found/
```
no builds are running. Hmm. The build info should be reset during reboot.

**Comment by frispete** [comment]
Okay, this issue seems to heal itself after some time.

Another observation, the statistical plots aren't working for me:
```
$ date
Tue Jul 16 11:00:06 UTC 2019
```
![image](https://user-images.githubusercontent.com/1009685/61289591-f1e39600-a7c9-11e9-8b42-a37cdd691849.png)

Stats for 1 hour timeframe are completely empty.

**Comment by seife** [comment]
> Stats for 1 hour timeframe are completely empty.

Check if the clockwork / delayed jobs are running (and if they are also scheduled to start after /srv/obs is mounted... ;-)

The statistics are created by the clockworkd / delayed-job thingy IIUC, and those are not monitored on the services monitor page.

**Comment by frispete** [comment]
I see, thanks again Stefan. Much appreciated.

obs-clockwork wasn't running and wasn't enabled. Duh.

**Comment by seife** [comment]
I think this is missing from the README.UPDATERS, tbh I'm not sure if I enabled it manually, but some manual work was needed afterwards.

**Comment by frispete** [comment]
OBS is up and running so far. With 80% zoom, it's fine, although I would prefer a more condensed layout.

What I'm really happy about, it's showing external links now!

**Comment by mgoppold** [comment]
in my setup (upgrade vom 2.9 on 42.3 to 2.10beta and 2.10 final on 15.1) the obs-clockwork.service wasn't enabled. Now my obs services looks like:

```
find /usr/lib/systemd/system -name obs\* -printf "echo -n '%f '; systemctl is-enabled %f\n"  | sort | bash
obsapisetup.service disabled
obs-api-support.target disabled
obs-clockwork.service enabled
obs-delayedjob-queue-consistency_check.service disabled
obs-delayedjob-queue-default.service disabled
obs-delayedjob-queue-issuetracking.service disabled
obs-delayedjob-queue-mailers.service disabled
obs-delayedjob-queue-project_log_rotate.service disabled
obs-delayedjob-queue-quick@.service disabled
obs-delayedjob-queue-releasetracking.service disabled
obs-delayedjob-queue-staging.service disabled
obsdeltastore.service enabled
obsdispatcher.service enabled
obsdodup.service enabled
obsgetbinariesproxy.service disabled
obsnotifyforward.service disabled
obspublisher.service enabled
obsredis.service disabled
obsrepserver.service enabled
obsscheduler.service enabled
obsservicedispatch.service enabled
obsservice.service enabled
obssigner.service enabled
obs-sphinx.service disabled
obssrcserver.service enabled
obsstoragesetup.service enabled
obswarden.service enabled
obsworker.service enabled
```
None of the `obs-delayedjob-queues` are enabled. Which one should enabled by default?

**Comment by hennevogel** [comment]
> Which one should enabled by default?

All of them.

**Comment by mgoppold** [comment]
And where should obs-delayedjob-queue-quick@.service point to?

**Comment by hennevogel** [comment]
point to?

**Comment by mgoppold** [comment]
@ Services point to somthing like openvpn@ to a net device, systemd-fsck@ to a block device.

Here we have:
```
ExecStart = /usr/bin/bundle.ruby2.5 exec script/delayed_job.api.rb --queue=quick start -i %i
```
with %i will replaced with ...?

**Comment by mgoppold** [comment]
OK %i is a worker from /usr/lib64/obs-api/ruby/2.5.0/gems/delayed_job-4.1.7/lib/delayed/command.rb you can see:

```
opt.on('-i', '--identifier=n', 'A numeric identifier for the worker.') do |n|
```
thus:

```
for worker in $(seq 1 8); do systemctl enable --now obs-delayedjob-queue-quick@$worker.service; done
```
with 8 workers.

**Comment by hennevogel** [comment]
yes, you can use one or more. the more workers you use the more jobs will be processed in parallel. To give you a perspective, on build.o.o we run 3 quick queues.

**Comment by mgoppold** [comment]
What makes a quick queue different from a normal worker? Is there any docu to read more?

**Comment by adrianschroeter** [comment]
The obsapisetup.service and obsstoragesetup.service should *NOT* be enabled in the packages, only in the appliance.

Also obsgetbinariesproxy, obsnotifyforward and obsredis makes little sense atm or without additional configuration.

**Comment by frispete** [comment]
> The obsapisetup.service and obsstoragesetup.service should NOT be enabled in the packages, only in the appliance.

If obsstoragesetup.service isn't enabled, what service mount `/srv/obs` then? If you look at my comment [above](https://github.com/openSUSE/open-build-service/issues/7912#issuecomment-511732404), `/etc/fstab` doesn't, which was the reason for this report in the first place. Or is obsstoragesetup.service supposed to setup /etc/fstab once, and being disabled thereafter?

**Comment by frispete** [comment]
> > Which one should enabled by default?

> All of them.

While at it, I found obs-api-support.target disabled as well, which gives a nice list of services, that might be missing after upgrade. Here is the list of services, that were missing here, in a format, that should fit README.UPDATERS:
```
$ systemctl enable --now \
 obs-clockwork.service \
 obs-delayedjob-queue-consistency_check.service \
 obs-delayedjob-queue-default.service \
 obs-delayedjob-queue-issuetracking.service \
 obs-delayedjob-queue-mailers.service \
 obs-delayedjob-queue-project_log_rotate.service \
 obs-delayedjob-queue-quick@0.service \
 obs-delayedjob-queue-releasetracking.service \
 obs-delayedjob-queue-staging.service \
 obs-sphinx.service \
 obs-api-support.target
```

**Comment by adrianschroeter** [comment]
On Mittwoch, 17. Juli 2019, 10:02:46 CEST Hans-Peter Jansen wrote:
> > The obsapisetup.service and obsstoragesetup.service should NOT be enabled
> > in the packages, only in the appliance.
> If obsstoragesetup.service isn't enabled, what service mount `/srv/obs`
> then? If you look at my comment
> [above](https://github.com/openSUSE/open-build-service/issues/7912#issuecom
> ment-511732404), `/etc/fstab` doesn't, which was the reason for this report
> in the first place. Or is obsstoragesetup.service supposed to setup
> /etc/fstab once, and being disabled thereafter?

When you use the plain packages it is up to the admin to define the storage.
He may use plain root disk or mount some external NAS there.

obstoragesetup and obsapisetup are supposed to be used with an appliance
or container in first place. Where you may replace the entire OS while
keep the production data. These service shall also take care the update 
situation then.

-- 

Adrian Schroeter
SUSE Linux Products GmbH,  Maxfeldstr. 5, 90409 Nuernberg, Germany
email: adrian@suse.de

**Comment by frispete** [comment]
Hmm, many moons ago, I followed an OBS installation guide, that resulted in this layout:
```
/dev/mapper/OBS-server         504G   47G  457G  10% /srv/obs
/dev/mapper/OBS-cache           25G   45M   24G   1% /var/cache/obs/worker
/dev/mapper/OBS-worker_root_1  122G   61M  116G   1% /var/cache/obs/worker/root_1
/dev/mapper/OBS-worker_root_2  122G   61M  116G   1% /var/cache/obs/worker/root_2
/dev/mapper/OBS-worker_root_3  122G   61M  116G   1% /var/cache/obs/worker/root_3
/dev/mapper/OBS-worker_root_4  122G   61M  116G   1% /var/cache/obs/worker/root_4
/dev/mapper/OBS-worker_root_5  122G   61M  116G   1% /var/cache/obs/worker/root_5
/dev/mapper/OBS-worker_root_6  122G   61M  116G   1% /var/cache/obs/worker/root_6
/dev/mapper/OBS-worker_root_7  122G   61M  116G   1% /var/cache/obs/worker/root_7
/dev/mapper/OBS-worker_root_8  122G   61M  116G   1% /var/cache/obs/worker/root_8
```

that is (re-)created from obsstoragesetup.service on every boot. It would be quite tedious to do manually, wouldn't it? (or am I miss something, I'm a device mapper noob)

**Comment by eduardoj** [comment]
> in my setup (upgrade vom 2.9 on 42.3 to 2.10beta and 2.10 final on 15.1) the obs-clockwork.service wasn't enabled. Now my obs services looks like:
> 
> ```
> find /usr/lib/systemd/system -name obs\* -printf "echo -n '%f '; systemctl is-enabled %f\n"  | sort | bash
> obsapisetup.service disabled
> obs-api-support.target disabled
> obs-clockwork.service enabled
> [...]
> ```
> 
> None of the `obs-delayedjob-queues` are enabled. Which one should enabled by default?

All of the `obs-delayedjob-queues` services should be disabled. `obs-delayedjob-queues` services, as well as [other services](https://github.com/openSUSE/open-build-service/blob/02d5f762a897bfee14dc7527440693266a0650db/dist/systemd/obs-api-support.target#L3) like `obs-clockwork` are managed by `obs-api-support.target`.

`obs-api-support.target` should be enabled, and in the upgrade from 2.9 to 2.10 there is [code](https://github.com/openSUSE/open-build-service/blob/02d5f762a897bfee14dc7527440693266a0650db/dist/obs-server.spec#L602) that enables `obs-api-support.target`.

Please enable `obs-api-support.target` and try again.

**Comment by frispete** [comment]
> obs-api-support.target should be enabled, and in the upgrade from 2.9 to 2.10 there is code that enables obs-api-support.target.

For some reason, that code didn't run here.

I reverted the changes made before, and just:

> Please enable obs-api-support.target and try again.
```
$ find /usr/lib/systemd/system -name obs\* -printf "echo -n '%f '; systemctl is-enabled %f\n"  | sort | bash
obs-api-support.target enabled
obs-clockwork.service disabled
obs-delayedjob-queue-consistency_check.service disabled
obs-delayedjob-queue-default.service disabled
obs-delayedjob-queue-issuetracking.service disabled
obs-delayedjob-queue-mailers.service disabled
obs-delayedjob-queue-project_log_rotate.service disabled
obs-delayedjob-queue-quick@.service disabled
obs-delayedjob-queue-releasetracking.service disabled
obs-delayedjob-queue-staging.service disabled
obs-sphinx.service disabled
obsapisetup.service disabled
obsdeltastore.service enabled
obsdispatcher.service enabled
obsdodup.service enabled
obsgetbinariesproxy.service disabled
obsnotifyforward.service disabled
obspublisher.service enabled
obsredis.service disabled
obsrepserver.service enabled
obsscheduler.service enabled
obsservice.service enabled
obsservicedispatch.service enabled
obssigner.service enabled
obssrcserver.service enabled
obsstoragesetup.service enabled
obswarden.service enabled
obsworker.service enabled
```

After rebooting, it still looks fine:
```
$ find /usr/lib/systemd/system -name obs\* -printf "echo -n '%f '; systemctl is-active %f\n"  | sort | bash
obs-api-support.target active
obs-clockwork.service active
obs-delayedjob-queue-consistency_check.service active
obs-delayedjob-queue-default.service active
obs-delayedjob-queue-issuetracking.service active
obs-delayedjob-queue-mailers.service active
obs-delayedjob-queue-project_log_rotate.service active
obs-delayedjob-queue-quick@.service Failed to retrieve unit state: Unit name obs-delayedjob-queue-quick@.service is neither a valid invocation ID nor unit name.
obs-delayedjob-queue-releasetracking.service active
obs-delayedjob-queue-staging.service active
obs-sphinx.service active
obsapisetup.service inactive
obsdeltastore.service active
obsdispatcher.service active
obsdodup.service active
obsgetbinariesproxy.service inactive
obsnotifyforward.service inactive
obspublisher.service active
obsredis.service inactive
obsrepserver.service active
obsscheduler.service active
obsservice.service active
obsservicedispatch.service active
obssigner.service active
obssrcserver.service active
obsstoragesetup.service active
obswarden.service active
obsworker.service active
```

Thanks @eduardoj

**Comment by mgoppold** [comment]
@frispete can you show me the output of `ls -l /etc/systemd/system/obs-api-support.target.wants`, please?

I have disables the services that obs-api-support.target wants  and enable obs-api-support.target but after reboot I run into https://github.com/openSUSE/open-build-service/issues/7912#issuecomment-511923582 state.

With `systemctl start obs-api-support.target` the services become active but not without this.

**Comment by frispete** [comment]
> This is about a bugreport for our code here or do you use this as support ticket system for your installation? ;)

Given, that this issue has helped the packman project at least already, I hope, it's not completely displaced.. 

Yes, I suffer from obvious knowledge deficits. E.g. I would love to find an answer for

> Or is obsstoragesetup.service supposed to setup /etc/fstab once, and being disabled thereafter?

But it's time to close this issue, since the major issue was resolved.

**Comment by hennevogel** [comment]
It was for you but not for other people.

**Comment by seife** [comment]
I'm not sure if this code can work.
Is the old package content still there in %post? If not,  the "systemctl is-enabled" will fail:
```
vsa2897004:~ # systemctl  is-enabled pcscd
indirect
vsa2897004:~ # echo $?
0
vsa2897004:~ # zypper rm pcsc-lite
vsa2897004:~ # systemctl  is-enabled pcscd
Failed to get unit file state for pcscd.service: No such file or directory
vsa2897004:~ # echo $?
1
```
(pcsd being a sysv init script that was just present on this machine ;-) )
So maybe you need to do this in %pre, create a marker file that then gets used in %post?

There certainly was a reason why the sysv -> systemd migration was put into macros ;-)

**Comment by mgoppold** [comment]
@hennevogel Thank you for reopening. 
@adrianschroeter Yes, we want to understand what went wrong during the Update while we following the README.UPDATERS. 

But so solve this we have to roll back and do the upgrade again ... ;-) I do not really want to do that.

My last open question is, and I can open a new issue if you want: Why don't start the necessary services at boot while  obs-api-support.target is enabled (and active)?

@frispete Unfortunately, you did not answer my question Here I have:

```
obs:~ # ls -l /etc/systemd/system/obs-api-support.target.wants
insgesamt 0
lrwxrwxrwx 1 root root 42 16. Jul 18:50 obs-sphinx.service -> /usr/lib/systemd/system/obs-sphinx.service
```
should not all wants be listed there?

I run
```
# systemctl enable --now obs-api-support.target
# systemctl is-active obs-clockwork.service obs-delayedjob-queue-consistency_check.service obs-delayedjob-queue-default.service obs-delayedjob-queue-issuetracking.service obs-delayedjob-queue-mailers.service obs-delayedjob-queue-project_log_rotate.service obs-delayedjob-queue-quick@0.service obs-delayedjob-queue-quick@1.service obs-delayedjob-queue-quick@2.service obs-delayedjob-queue-releasetracking.service obs-delayedjob-queue-staging.service obs-sphinx.service | sort | uniq -c
     12 active
# reboot

# systemctl is-active obs-clockwork.service obs-delayedjob-queue-consistency_check.service obs-delayedjob-queue-default.service obs-delayedjob-queue-issuetracking.service obs-delayedjob-queue-mailers.service obs-delayedjob-queue-project_log_rotate.service obs-delayedjob-queue-quick@0.service obs-delayedjob-queue-quick@1.service obs-delayedjob-queue-quick@2.service obs-delayedjob-queue-releasetracking.service obs-delayedjob-queue-staging.service obs-sphinx.service | sort | uniq -c
      2 active
     10 failed
# systemctl status obs-api-support.target
● obs-api-support.target - Open Build Service API Support Daemons
   Loaded: loaded (/usr/lib/systemd/system/obs-api-support.target; enabled; vendor preset: disabled)
   Active: active since Thu 2019-07-18 20:12:09 UTC; 21s ago

Jul 18 20:12:09 obs systemd[1]: Reached target Open Build Service API Support Daemons.
# systemctl start obs-api-support.target
# systemctl is-active obs-clockwork.service obs-delayedjob-queue-consistency_check.service obs-delayedjob-queue-default.service obs-delayedjob-queue-issuetracking.service obs-delayedjob-queue-mailers.service obs-delayedjob-queue-project_log_rotate.service obs-delayedjob-queue-quick@0.service obs-delayedjob-queue-quick@1.service obs-delayedjob-queue-quick@2.service obs-delayedjob-queue-releasetracking.service obs-delayedjob-queue-staging.service obs-sphinx.service | sort | uniq -c
     12 active
```
Mybe it is a dependicy issue.

**Comment by mgoppold** [comment]
> It was for you but not for other people. We still don't know why the [code](https://github.com/openSUSE/open-build-service/blob/02d5f762a897bfee14dc7527440693266a0650db/dist/obs-server.spec#L602) in our spec file did not succeed for you, packman or @mgoppold

The target obs-api-support.target will only be activated if obsapidelayed.service is-enabled before.  Is the assumption correct during the dist+package upgrade from obs appilance 2.9 on Leap 42.3 to obs 2.10 on Leap 15.1?   

If the obsapidelayed.service on 2.9 is startet by another service/target how obs-api-support.target  should do that now then the obs-api-support.target  will not activated during update.

**Comment by eduardoj** [comment]
@seife:

> I'm not sure if this code can work.
> [..]
> So maybe you need to do this in %pre, create a marker file that then gets used in %post?
> 
> There certainly was a reason why the sysv -> systemd migration was put into macros ;-)

This upgrade was tested manually and worked, before the code was merged. And tested again manually as this issue arose. And the test was to upgrade an OBS appliance 2.9 with `obsapidelayed.service` enabled to OBS 2.10, and check that `obs-api-support.target` is enabled after that.

@mgoppold:

> The target obs-api-support.target will only be activated if obsapidelayed.service is-enabled before. Is the assumption correct during the dist+package upgrade from obs appilance 2.9 on Leap 42.3 to obs 2.10 on Leap 15.1?

We can say "obs-api-support.target will only be **enabled** if obsapidelayed.service is-enabled before".

> If the obsapidelayed.service on 2.9 is startet by another service/target how obs-api-support.target should do that now then the obs-api-support.target will not activated during update.

`obsapidelayed.service` on OBS 2.9 should be enabled in a "I followed the docs" installation, or upgrade, of OBS. If by any chance `obsapidelayed.service` is not enabled in 2.9, `obs-api-support.target` should not be enabled in the upgrade.

I close this issue. Please fell free to open it again if happens again.

**Comment by frispete** [comment]
As far as I can tell, if coming from an earlier installation, you probably arrive at 2.9 with obsapidelayed.service disabled. Since OBS 2.9 works fine without, nobody cared. With 2.10 and obs-api-support.target disabled, OBS behaves sub-optimal at best, therefor

* it needs a related note in README.UPDATERS
* or even being eabled unconditionally

for best UX (all put under the clause: in my **humble** opinion, of course).
